sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -42,9 +516,171 @@ export class StreamingReceiver extends MessageReceiver {
+      );
+      return;
+    }
+
+    this._isDetaching = true;
+    try {
+      // Clears the token renewal timer. Closes the link and its session if they are open.
+      // Removes the link and its session if they are present in rhea's cache.
+      await this._closeLink(this._receiver);
+
+      if (this.receiverType === ReceiverType.batching) {
+        log.error(
+          "[%s] Receiver '%s' with address '%s' is a Batching Receiver, so we will not be " +
+            "re-establishing the receiver link.",
+          connectionId,
+          this.name,
+          this.address
+        );
+        return;
+      }
+
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // Track-1
+      //   - We should only attempt to reopen if either no error was present,
+      //     or the error is considered retryable.
+      // Track-2
+      //  Reopen
+      //   - If no error was present
+      //   - If the error is a MessagingError and is considered retryable
+      //   - Any non MessagingError because such errors do not get
+      //     translated by `@azure/core-amqp` to a MessagingError
+      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
+      const shouldReopen =
+        translatedError instanceof MessagingError ? translatedError.retryable : true;
+
+      // Non-retryable errors that aren't caused by disconnect
+      // will have already been forwarded to the user's error handler.
+      // Swallow the error and return quickly.
+      if (!shouldReopen && !causedByDisconnect) {
+        log.error(
+          "[%s] Encountered a non retryable error on the receiver. Cannot recover receiver '%s' with address '%s' encountered error: %O",
+          connectionId,
+          this.name,
+          this.address,
+          translatedError
+        );
+        return;
+      }
+
+      // Non-retryable errors that are caused by disconnect
+      // haven't had a chance to show up in the user's error handler.
+      // Rethrow the error so the surrounding try/catch forwards it appropriately.
+      if (!shouldReopen && causedByDisconnect) {
+        log.error(
+          "[%s] Encountered a non retryable error on the connection. Cannot recover receiver '%s' with address '%s': %O",
+          connectionId,
+          this.name,
+          this.address,
+          translatedError
+        );
+        throw translatedError;
+      }
+
+      // shall retry forever at an interval of 15 seconds if the error is a retryable error
+      // else bail out when the error is not retryable or the operation succeeds.
+      const config: RetryConfig<void> = {
+        operation: () =>
+          this.init(
+            // provide a new name to the link while re-connecting it. This ensures that
+            // the service does not send an error stating that the link is still open.
+            true,
+            undefined
```

Comment:
Isnt the second argument optional here?
```suggestion
            true
```

Reply:
Yes, removed.